File: WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/PostsListPage.java
package org.wordpress.android.e2e.pages;

import android.view.ViewGroup;

import androidx.recyclerview.widget.RecyclerView;
import androidx.viewpager.widget.ViewPager;

import org.wordpress.android.R;

import static org.wordpress.android.support.WPSupportUtils.getCurrentActivity;
import static org.wordpress.android.support.WPSupportUtils.getTranslatedString;
import static org.wordpress.android.support.WPSupportUtils.scrollToAndClickOnTextInRecyclerView;
import static org.wordpress.android.support.WPSupportUtils.scrollToTopOfRecyclerView;
import static org.wordpress.android.support.WPSupportUtils.selectItemWithTitleInTabLayout;

public class PostsListPage {
    public PostsListPage() {}

    public static void tapPostWithName(String name) {
        scrollToAndClickOnTextInRecyclerView(name, getRecyclerView());
    }

    public static void scrollToTop() {
        scrollToTopOfRecyclerView(getRecyclerView());
    }

    private static RecyclerView getRecyclerView() {
        ViewPager pager = getCurrentActivity().findViewById(R.id.postPager);
        RecyclerView recyclerView = (RecyclerView) pager.getChildAt(pager.getCurrentItem())
                                                        .findViewById(R.id.recycler_view);
        if (recyclerView == null) {
            // Workaround for cases when recyclerview id missing
            recyclerView = (RecyclerView) ((ViewGroup) ((ViewGroup) (pager.getChildAt(pager.getCurrentItem()))
                    .findViewById(R.id.ptr_layout)).getChildAt(0)).getChildAt(0);
        }
        return recyclerView;
    }

    public static void goToDrafts() {
        selectItemWithTitleInTabLayout(getTranslatedString(R.string.post_list_tab_drafts), R.id.tabLayout);
    }
}

File: WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/PostsListPage.kt
package org.wordpress.android.e2e.pages

import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import androidx.viewpager.widget.ViewPager
import org.wordpress.android.R
import org.wordpress.android.support.WPSupportUtils

object PostsListPage {
    @JvmStatic
    fun tapPostWithName(name: String?) {
        WPSupportUtils.scrollToAndClickOnTextInRecyclerView(name, recyclerView)
    }

    @JvmStatic
    fun scrollToTop() {
        WPSupportUtils.scrollToTopOfRecyclerView(recyclerView)
    }

    // Workaround for cases when recyclerview id missing
    private val recyclerView: RecyclerView?
        private get() {
            val pager = WPSupportUtils.getCurrentActivity().findViewById<ViewPager>(R.id.postPager)
            var recyclerView = pager.getChildAt(pager.currentItem)
                .findViewById<View>(R.id.recycler_view) as RecyclerView
            if (recyclerView == null) {
                // Workaround for cases when recyclerview id missing
                recyclerView = ((pager.getChildAt(pager.currentItem)
                    .findViewById<View>(R.id.ptr_layout) as ViewGroup).getChildAt(0) as ViewGroup).getChildAt(
                    0
                ) as RecyclerView
            }
            return recyclerView
        }

    @JvmStatic
    fun goToDrafts() {
        WPSupportUtils.selectItemWithTitleInTabLayout(
            WPSupportUtils.getTranslatedString(R.string.post_list_tab_drafts),
            R.id.tabLayout
        )
    }
}
